src/main/java/com/netgrif/application/engine/migration/RuntimeMigrationAspect.java

```diff
@@ -0,0 +1,70 @@
+package com.netgrif.application.engine.migration;
+
+import com.netgrif.application.engine.petrinet.domain.PetriNet;
+import com.netgrif.application.engine.petrinet.domain.dataset.FieldType;
+import com.netgrif.application.engine.petrinet.domain.dataset.UserFieldValue;
+import com.netgrif.application.engine.petrinet.domain.dataset.UserListFieldValue;
+import org.aspectj.lang.annotation.Aspect;
+import org.bson.Document;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Order
+@Aspect
+@Component
+public class RuntimeMigrationAspect extends AbstractRuntimeMigrationAspect {
+
+    @Override
+    public void callProcessMigrations(Document bson) {
+
+    }
+
+    @Override
+    public void callCaseMigrations(Document bson) {
+        migrateUserListFieldValueType(bson);
+    }
+
+    @Override
+    public void callTaskMigrations(Document bson) {
+
+    }
+
+    protected final void migrateUserListFieldValueType(Document document) {
```

Comment:
This method is too slow to be called on every fetched case. It uses too many calls.
It is planned for migration framework rework in the future.